pkg/function/main_test.go

```diff
@@ -38,6 +38,32 @@ func ExampleInputs() {
 	// Output: {"apiVersion":"config.kubernetes.io/v1","kind":"ResourceList","items":[{"metadata":{"creationTimestamp":null,"name":"foobar\n"},"spec":{"containers":null},"status":{}}]}
 }
 
+func ExampleAddCustomInputType() {
+	type myType struct {
+		SecretKey string
+	}
+
+	AddCustomInputType(func(in *corev1.Secret) (*myType, error) {
```

Comment:
are they always secrets or sometimes configmap with data?

Reply:
the func is generic so anything that implements client.Object should work!